editor/src/components/inspector/sections/component-section/component-section.tsx

```diff
@@ -1142,7 +1128,59 @@ export class ComponentSection extends React.Component<
   }
 }
 
-function useVariablesInScopeForSelectedElement() {
+interface VariableOption {
+  variableName: string
+  definedElsewhere: string | null
+  value: string
+}
+
+function valuesFromObject(
+  name: string,
+  objectName: string,
+  value: object | null,
+): Array<VariableOption> {
+  if (value == null) {
+    return [{ variableName: name, definedElsewhere: null, value: `null` }]
+  }
+
+  const patchDefinedElsewhereInfo = (variable: VariableOption) => ({
+    variableName: variable.variableName,
+    value: variable.value,
+    definedElsewhere: objectName,
+  })
+
+  if (Array.isArray(value)) {
+    return value.flatMap((v, idx) =>
+      valuesFromVariable(`${name}[${idx}]`, v).map((variable) =>
+        patchDefinedElsewhereInfo(variable),
+      ),
+    )
+  }
+
+  return Object.entries(value).flatMap(([key, field]) =>
+    valuesFromVariable(`${name}.${key}`, field).map((variable) =>
```

Comment:
beware of objects like this `{['I am legal JS']: 'beware of me'}`, I _think_ this code would produce `objectName.I am legal JS` as the code

Reply:
I'll refactor it to `objectName['I am legal JS']`

Reply:
https://github.com/concrete-utopia/utopia/pull/4858/commits/602f3c9c6c3c97b30d43d27be1b59c95983c1f74